app/src/main/java/edu/cascadia/mobile/apps/movies/EditMovie.java

```diff
@@ -24,32 +26,66 @@ protected void onCreate(Bundle savedInstanceState) {
+
+        //use -1 as a flag for a movie not yet in the database
+        final int mId = intent.getIntExtra("movieId",-1);
+
+        //if the movie was found in the DB...
+        if (mId != -1) {
+            //get current movie
+            MovieEntity selectedMovie = mDatabase.movieDao().getMovie(mId);
+
+            //get and set existing movie data in fragment EditText views
+            ((EditText)findViewById(R.id.edit_title)).setText(selectedMovie.getTitle());
+            ((EditText)findViewById(R.id.edit_director)).setText(selectedMovie.getDirector());
+            ((EditText)findViewById(R.id.edit_year)).setText(selectedMovie.getYear());
+            ((EditText)findViewById(R.id.edit_runtime)).setText(Integer.toString(selectedMovie.getRunTime()));
+        }
+
+
         FloatingActionButton fab = findViewById(R.id.fab);
         fab.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
                 //TODO manage movie update by setting movie id if avaliable
-                int movieId = 0;
 
-                //Add Movie
+                //data variables
                 EditText eTitle = findViewById(R.id.edit_title);
                 EditText eDirector = findViewById(R.id.edit_director);
                 EditText eYear = findViewById(R.id.edit_year);
                 EditText eRuntime = findViewById(R.id.edit_runtime);
+
                 int runtime = Integer.parseInt(eRuntime.getText().toString());
 
-                MovieEntity mMovie = new MovieEntity(
-                        movieId,
-                        eTitle.getText().toString(),
-                        eDirector.getText().toString(),
-                        eYear.getText().toString(),
-                        runtime
-                        );
+                MovieEntity mMovie;
 
-                mDatabase.movieDao().addOrUpdate(mMovie);
+                //handle new movie or existing case id
+                boolean isNew = (mId == -1);
+                String snackBarMessage;
 
+                if (isNew) {
+                    mMovie = new MovieEntity(
+                            (mDatabase.movieDao().getMaxId() + 1),
+                            eTitle.getText().toString(),
+                            eDirector.getText().toString(),
+                            eYear.getText().toString(),
+                            runtime
+                    );
+                    mDatabase.movieDao().addOrUpdate(mMovie);
+                    snackBarMessage = "Added New Record: " + mMovie.getTitle();
+                }
+                else {
+                    mMovie = mDatabase.movieDao().getMovie(mId);
+                    mMovie.setTitle(eTitle.getText().toString());
+                    mMovie.setDirector(eDirector.getText().toString());
+                    mMovie.setYear(eYear.getText().toString());
+                    mMovie.setRunTime(Integer.parseInt(eRuntime.getText().toString()));
```

Comment:
This is the section of code dealing with editing an existing movie information.
Once you finish setting all of its properties here, you need to use the Dao to update the database, the same as with the new movie in line 76 above.
`mDatabase.movieDao().addOrUpdate(mMovie);`

the `addOrUpdate()` method has internal logic to update the database record if its movie parameter has its Id property set.